src/inc/corprof.idl

```diff
@@ -2408,6 +2412,31 @@ interface ICorProfilerCallback8 : ICorProfilerCallback7
         [in] BOOL       fIsSafeToBlock);
 }
 
+[
+    object,
+    uuid(27583EC3-C8F5-482F-8052-194B8CE4705A),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback9 : ICorProfilerCallback8
+{
+    // This event is triggered whenever a dynamic method is garbage collected
+    // and subsequently unloaded.
+
+    HRESULT DynamicMethodUnloaded([in] FunctionID functionId);
+
+    HRESULT EventPipeEventDelivered(
+        [in] REFGUID eventGuid,
```

Comment:
Brian may also bring this up (because I pushed it on him as well), I would like to avoid GUIDs as a way of identifying providers.   I would like this to be a string instead.   Brian will change the EventPipe code for this, but you should do it as well (at worst, just string-ify the GUID, but we should be able to provide you with the Provider name).